### ISSUE TITLE
Rewrite wish list 

### ISSUE THREAD

**Issue by michaelsutton** [kaspanet/rusty-kaspa#15]
An issue for gathering any possible feature wished to be implemented during the rewrite of Kaspa to Rust. This might help guiding the developers to design things in a specific way, even if the actual feature is yet to be implemented.

**Comment by Aleoami** [comment]
The option to let getting UTXOs being spent from "all addresses except this this and this".

**Comment by KaffinPX** [comment]
Ditch gRPC on RPC and using JSON-HTTP as main provider.

**Comment by KaffinPX** [comment]
Planned RPC endpoints, not a chaos like go-kaspad.

**Comment by someone235** [comment]
txindex

**Comment by D-Stacks** [comment]
- eUTXO model.

**Comment by D-Stacks** [comment]
- base58 encoded addresses

**Comment by D-Stacks** [comment]
- benchmark msgpack alternative to gRPC in both p2p and rpc perhaps for future switch, if indeed more preformant.

**Comment by D-Stacks** [comment]
- developer wallet mode, which unlocks among potentially other things, payloads and custom signing operations.

**Comment by D-Stacks** [comment]
Archival subnetwork:

- Allow the concept of partial archival nodes:

1) allow for a random section of the blockdag to be stored by nodes (for example with a `--archival 3`), which allows the network to distribute the archival functionality among various kaspads offering diskspace, without commiting to the full blockDAG ledger. I guess the number should reffer to the number of prunning periods that the kaspad will store.

2) allow for partial archival nodes with a certain lookback period (for example with a `--archival-lookback 3`) which will store 3 extra pruning points beyond the current minimum.

**Comment by XVGHypeMan** [comment]
This is like a side note - even though Coinbase Rosetta is a separate entity, it may be wise if its not too difficult to provide mechanisms to answer the calls it makes etc.

https://www.rosetta-api.org/docs/Reference.html

**Comment by KaffinPX** [comment]
> * eUTXO model.

Payloads in outputs*.

**Comment by KaffinPX** [comment]
Better call names (on RPC)?

**Comment by michaelsutton** [comment]
No crashes whatsoever.

**Comment by D-Stacks** [comment]
> Compare(benchmark) ed25519(blake2b) with secp256k1 for a possible switch in future.

I don't think this would be a switch though, since need backward compatibility, but can support ed25519 ontop.

**Comment by D-Stacks** [comment]
- certificates, proxies and support tor addresses.

**Comment by Aleoami** [comment]
Native pagination of the UTXO data to avoid timeouts on a mega-multi-UTXO addresses

**Comment by D-Stacks** [comment]
simple ui display ports for kaspad and kaspawalletd.

**Comment by D-Stacks** [comment]
Stratum implementation on node level.

This can really come in varying degrees / with suboptions:
- 1) Var diff so miners can display hashrate and better mining stats.
- 2) Seperate Api for stats

**Comment by D-Stacks** [comment]
Better automatic utxo selection scheme in wallet: perhaps dynamic based on utxostate size?

**Comment by michaelsutton** [comment]
> Stratum implementation on node level.
> 
> This can really come in varying degrees / with suboptions:
> 
> * 1. Var diff so miners can display hashrate and better mining stats.
> * 2. Seperate Api for stats

Cant it be based on https://github.com/fungibilly/kaspad-stratum, given that it is written in rust ?

**Comment by D-Stacks** [comment]
> > Stratum implementation on node level.
> > This can really come in varying degrees / with suboptions:
> > 
> > * 1. Var diff so miners can display hashrate and better mining stats.
> > * 2. Seperate Api for stats
> 
> Cant it be based on https://github.com/fungibilly/kaspad-stratum, given that it is written in rust @D-Stacks ?

based probably, but i know it has some issues, so no miner is actually using it. Also it would need to be expanded / altered to allow for multiple solo mining participants. 

Currently adapters get told on start-up to which address they should mine too. In the case of a stratum on the node it should be inferred by `mining.authorize` method, and map the building of blocktemplates to corresponding connections.

Beyond this I would think about implementing Var diff for hashrate stat collection, one of he main reasons the adapters aren't used is 

1) extra installation steps  - I suppose having it built into the node completely negates this
2) miners mine to pools to receive information about their mining statistics

If the idea is to decentralize mining power as much as possible towards to node mining, this would greatly help. currently we have over 56%+ of miners pool mining, despite having 3 open source stratum adapters available.  Ideally i'd even add a ui port where miners can see data, but so far this seems beyond the scope of the node.

**Comment by D-Stacks** [comment]
- Adopt bitcoin cash op codes standard.

**Comment by Aleoami** [comment]
Node RPC subsystem stats request

**Comment by D-Stacks** [comment]
Fast and optimized client wallet that can be scaled for high transactional rates, and automated usages i.e. pools, exchanges and others.

- multiple outputs in wallet send (also in transaction outputs).
- async requests
- ...

**Comment by D-Stacks** [comment]
benchmark sustainable min. gas limit per block to in accordance with min. requirements to run a node.

**Comment by freshair18** [comment]
@michaelsutton @someone235   I believe we can close this issue. Objections?
